### DISCUSSION TITLE
Init tensor with device agnostic code without it showing up in the state dict

### DISCUSSION THREAD

**Discussion by Robinysh** [Lightning-AI/pytorch-lightning#8450] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
This is the [current recommended way](https://pytorch-lightning.readthedocs.io/en/stable/advanced/multi_gpu.html#init-tensors-using-type-as-and-register-buffer) for initializing tensors inside LightningModules with device-agnostic code.
```
class LitModel(LightningModule):

    def __init__(self):
        ...
        self.register_buffer("sigma", torch.eye(3))
        # you can now access self.sigma anywhere in your module
```
Unfortunately, for my current use case, the tensor now shows up in the model state dict as an undesired side effect. My pipeline consists of loading the save file across runs, and sometimes some particular tensors change sizes, which makes my savefile incompatible. I don't want to save and load those particular tensors. Is there a way to work around this?

**Comment by tchaton** [comment]
Dear @Robinysh,

You could override the state_dict function of your LightningModule and drop those tensors.

Best,
T.C
